### PR TITLE
Rewrite Appointment List e2e Test in Cypress

### PR DESCRIPTION
**Pull request by narin**
## Description
Explore Cypress test runner by re-writing appointment list e2e test.

I wanted to import and reuse some of the helper functions in vaos-helpers.js but I was running into a strange error when trying to import it.

Also I was unable to implement the axe checks in these tests as there seems to be an issue with our axe plugin documented in the [Cypress migration guide](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/teams/vsp/teams/tools/frontend/cypress-migration-guide.md).

## Testing done
E2E

## Screenshots
Screen Shot 2020-06-22 at 10 39 43 PM

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/vaos/tests/e2e/vaos-cypress-helpers.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+            date: '2019-11-11T12:26:13.931+0000',
+            assigningAuthority: 'ICN',
+            systemId: 'var',
+          },
+        },
+      ],
+    },
+  });
+
+  cy.visit('health-care/schedule-view-va-appointments/appointments/');
```
Comment (jbalboni) [inline src/applications/vaos/tests/e2e/vaos-cypress-helpers.js:176]: Seems odd to have these two lines inside a function called `initAppointmentListMock`. Maybe move them to the main test file?

**Review by jbalboni — APPROVED** [review]
Some minor comments, but this looks great.

**Review by U-DON — APPROVED** [review]

File: src/applications/vaos/tests/e2e/appointment-list.cypress.spec.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      .first()
+      .click();
+
+    cy.get('#cancelAppt .usa-button').click();
+
+    cy.get('#cancelAppt button:not([disabled])')
+      .first()
+      .click();
+
+    cy.get('#cancelAppt').should('not.exist');
```
[the remark this answers is not in the document]
Comment (narin) [reply src/applications/vaos/tests/e2e/appointment-list.cypress.spec.js:23]: Good catch.  Added.

File: src/applications/vaos/tests/e2e/vaos-cypress-helpers.js (hunk 1 above)
Comment (narin) [reply src/applications/vaos/tests/e2e/vaos-cypress-helpers.js]: Moved these back into the test file.

File: src/applications/vaos/tests/e2e/vaos-cypress-helpers.js [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+  return data;
+}
+
+export function initAppointmentListMock() {
+  cy.server();
+  cy.login();
+  cy.route({
+    method: 'GET',
+    url: '**/v0/feature_toggles*',
```
[the remark this answers is not in the document]
Comment (narin) [reply src/applications/vaos/tests/e2e/vaos-cypress-helpers.js]: Thanks.  I've removed the unneeded wildcards are the beginning.  The only ones I'm using regex for now are the routes that have query params.

File: src/applications/vaos/tests/e2e/vaos-cypress-helpers.js [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+import pact from '../../api/pact.json';
+import confirmedVA from '../../api/confirmed_va.json';
+import confirmedCC from '../../api/confirmed_cc.json';
+import requests from '../../api/requests.json';
+import cancelReasons from '../../api/cancel_reasons.json';
+import supportedSites from '../../api/sites-supporting-var.json';
+import facilities from '../../api/facilities.json';
+import facilities983 from '../../api/facilities_983.json';
+import clinicList983 from '../../api/clinicList983.json';
+import slots from '../../api/slots.json';
```
[the remark this answers is not in the document]
Comment (narin) [reply src/applications/vaos/tests/e2e/vaos-cypress-helpers.js]: Good to know, thanks!

**Pull request merged**